src/lexer.coffee

```diff
@@ -1252,7 +1252,15 @@ REGEX = /// ^
 REGEX_FLAGS  = /^\w*/
 VALID_FLAGS  = /^(?!.*(.).*\1)[imguy]*$/
 
-HEREGEX      = /// ^(?: [^\\/#] | \\[\s\S] | /(?!//) | \#(?!\{) )* ///
+HEREGEX      = /// ^
+      (?:
+        [^\\/#]   |
+        \\[\s\S]  |
+        # The comment should consume everything until the end of the line, including the '///'
+        \#(?!\{)(?:[^\n\r]*[\n\r])  |
```

Comment:
As far as I remember, the lexer starts by normalizing all newlines, so only checking for `\n` should be enough.

https://github.com/jashkenas/coffeescript/blob/04959162c64d7e4212ffcb90026aec5a5b059ee2/src/lexer.coffee#L94

Reply:
:thumbsup: